py/desidatamodel/render.py

```diff
@@ -0,0 +1,63 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""
+====================
+desidatamodel.render
+====================
+
+Render the column descriptions file.
+"""
+import csv
+import sys
+import importlib.resources as ir
+import jinja2
+
+
+def format_columns(rows):
+    """Does something.
+
+    Parameters
+    ----------
+    rows : iterable
+        An iterable containing rows with any number of columns.
+
+    Returns
+    -------
+    :class:`tuple`
+        A tuple containing a format string, and an RST-style table separator.
+    """
+    lengths = list()
+    for row in rows:
+        for k, col in enumerate(row):
+            try:
+                if len(col) > lengths[k]:
+                    lengths[k] = len(col)
+            except IndexError:
+                lengths.append(len(col))
+    format_string = " ".join(["{{{0}:{1}}}".format(k, c) for k, c in enumerate(lengths)])
+    separator = ' '.join(['='*k for k in lengths])
+    return (format_string, separator)
+
+
+def main():
+    """Entry point for command-line scripts.
+
+    Returns
+    -------
+    :class:`int`
+        An integer suitable for passing to :func:`sys.exit`.
+    """
+    env = jinja2.Environment(loader=jinja2.PackageLoader('desidatamodel', package_path='data'),
+                             trim_blocks=True)
+    template = env.get_template('column_descriptions.rst')
+    columns = ir.files('desidatamodel') / 'data' / 'column_descriptions.csv'
```

Comment:
Although this syntax is kinda nifty, it's also a bit obscure and relies upon knowing that `ir.files(...)` returns a `pathlib.PosixPath` (not str!) that overrides the division operator to allow chaining paths and isn't just a typo.  

Consider updating this (and elsewhere) to the more familiar `os.path.join(ir.files('desidatamodel'), 'data', 'column_descriptions.csv')` or the less portable but also less verbose and perhaps even more familiar `str(ir.files('desidatamodel')) + "/data/column_descriptions.csv"` or `f"{ir.files('desidatamodel')}/data/column_descriptions.csv"` .  I admit that the PosixPath version is more readable once you know what it means, but that's the rub -- "once you know what it means..."

Unfortunately (and somewhat bizarrely), PosixPath supports `/` to chain strings, but not `+`.

It's not that big a deal to me here in this detailed function of desidatamodel, but I'd prefer that we *not* propagate this usage of PosixPath string chaining when we convert desispec et al to use `importlib.resources.files` instead of `pkg_resources.resource_filename` so I thought I'd mention it here.

Reply:
I disagree with the characterization of the notation as 'obscure'.  `pathlib` has been around for a long time (Python 3.4).  It may be "unfamiliar" to a subset of people who have not yet transitioned to using more modern Python path notation.

In particular, [this migration guide](https://importlib-resources.readthedocs.io/en/latest/migration.html) assumes that `pathlib` is familiar to developers.

That said, if you would prefer to not use `pathlib` in other packages, that's fine.

Reply:
Thanks for pointing to that migration guide.  I might warm up to the PosixPath overload `/` syntax someday...

Reply:
In the interest of moving this PR forward, I'm going to leave the existing code here as-is, but I take your point about not propagating this to other packages without further discusion.